packages/unmock-node/src/backend/index.ts

```diff
@@ -29,168 +32,159 @@ const mHttp = (
-                },
-              );
-              // https://github.com/nodejs/node/blob/master/lib/_http_client.js
-              // the original res.on('end') has a closure that refers to this
-              // as far as i can understand, 'this' is supposed to refer to res
-              f.apply(res, [d]);
-            },
-          ]);
-        }
-        return protoOn.apply(res, [s, f]);
+    return;
+  }
+  // tslint:disable-next-line:max-line-length
+  const pathForFake = buildPath(
+    rawHeaders,
+    h,
+    p,
+    ignore,
+    req.method,
+    req.url,
+    signature,
+    story.story,
+    unmockHost,
+    token !== undefined,
+  );
+  const doEndReporting = (fromCache: boolean) =>
+    (hash: string, requestBody: Buffer[], responseHeaders: any, responseBody: Buffer[]) => endReporter(
+    hash,
+    logger,
+    persistence,
+    save,
+    story.story,
+    token !== undefined,
+    fromCache,
+    {
+      lang: "node",
+    },
+    {
+      ...(requestBody.length > 0 ? { body: requestBody.map((buffer) => buffer.toString()).join("")} : {}),
+      headers: rawHeaders,
+      host: h,
+      method: req.method,
+      path: req.url,
+    },
+    {
+      headers: responseHeaders,
+      ...(responseBody.length > 0 ? { body: responseBody.map((buffer) => buffer.toString()).join("")} : {}),
+    },
+  );
+  const unmockHeaders = {
+    [UNMOCK_UA_HEADER_NAME]: JSON.stringify("node"),
+    ...(token ? {Authorization: `Bearer ${token}`} : {}),
+  };
+  passthrough(
+    req,
+    res,
+     ({
+      body,
+      headers,
+      host,
+      method,
+      path,
+     }) => {
+      const hashable = {
+        body: body.length > 0 ? body.map((buffer) => buffer.toString()).join("") : {},
+        headers,
+        hostname: host,
+        method,
+        path,
+        story: story.story,
+        user_id: userId ? userId : MOSES,
+        ...(signature ? {signature} : {}),
       };
-      if (second && typeof second === "function") {
-        (second as ((res: IncomingMessage) => void))(res);
-      } else if (third && typeof third === "function") {
-        (third as ((res: IncomingMessage) => void))(res);
+      const hash = v0(hashable, ignore);
+      const hasHash = persistence.hasHash(hash);
+      const makesNetworkCall = mode === Mode.ALWAYS_CALL_UNMOCK ||
+        (mode === Mode.CALL_UNMOCK_FOR_NEW_MOCKS && !hasHash);
+      snapshot({
+        hash,
+        host,
+        method,
+        path,
+      });
+      if (!makesNetworkCall) {
+        const response = persistence.loadResponse(hash);
+        Object.entries(response.headers).forEach(([k, v]) => {
+          res.setHeader(k, `${v}`);
+        });
+        res.write(response.body);
+        res.end();
+        doEndReporting(true)(hash, body, response.headers, !response.body ? [] : [Buffer.from(response.body)]);
+        return {
+          body,
+          hash,
+          headers,
+          host,
+          intercepted: true,
```

Comment:
Is `intercepted` effectively `fromCache`?

Reply:
mmm... yes